plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/sqs.ts

```diff
@@ -67,9 +68,13 @@ export class SqsServiceExtension implements ServiceExtension {
           spanAttributes[SemanticAttributes.MESSAGING_OPERATION] =
             MessagingOperationValues.RECEIVE;
 
-          request.commandInput.MessageAttributeNames = (
-            request.commandInput.MessageAttributeNames ?? []
-          ).concat(propagation.fields());
+          const messageAttributeNames = request.commandInput.MessageAttributeNames ?
+          deduplicateMessageAttributeNames(request.commandInput.MessageAttributeNames) : [];
+
+          request.commandInput.MessageAttributeNames = [
+            ...messageAttributeNames,
+            ...propagation.fields()
+          ];
         }
```

Comment:
Now that we have a new function in `MessageAttributes.ts` to handle this task, what do you think about moving this logic into this function as well? As this code handles message attribute (names) I would expect it to be part of this file

Reply:
There is other logic within the `sqs.ts` file that also handles MessageAttributes (e.g. line 110 in sqs.ts), so we were following the pattern set there. We're not sure that reassigning in the other file would improve the logic or readability.